tests/src/rules/namespace.js

```diff
@@ -1,5 +1,5 @@
 import { test, SYNTAX_CASES, getTSParsers, testVersion, testFilePath, parsers } from '../utils';
-import { RuleTester } from 'eslint';
+import { FlatCompatRuleTester as RuleTester } from '../rule-tester';
 import flatMap from 'array.prototype.flatmap';
 
 const ruleTester = new RuleTester({ env: { es6: true } });
```

Comment:
This line is where the `env` testing error is coming from in the CI failures.  Looks to be the only test that passes in an `env`.

Reply:
correct - I've resolved that locally by replacing it with setting the `ecmaVersion` since mapping `env` to `globals` has turned out to be very "interesting" especially as the latest version of the `globals` package requires Node v18+.

Next issue is that ESLint doesn't want to treat the imported files by this test as a `module`:

```
      AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: Should have no errors but had 1: [
  {
    ruleId: 'rule-to-test/namespace',
    severity: 1,
    message: "Parse errors in imported module './named-exports': sourceType 'module' is not supported when ecmaVersion < 2015. Consider adding `{ ecmaVersion: 2015 }` to the parser options. (undefined:undefined)",
    line: 1,
    column: 24,
    nodeType: 'Literal',
    endLine: 1,
    endColumn: 41
  }
]
```

Reply:
we don't need to use the latest version of the `globals` package tho.

Reply:
I've not looked into it too deeply yet, but I believe the way globals work in v9+ has changed to require strings instead of booleans which means older versions of `globals` don't work out of the box + I expect that we might need to use a very old version of the package for Node v4 support.

Ultimately right now I've gone this route because as @michaelfaith said this seems to be the only test that actually uses `env` and looks like it's only doing so for the ecma version rather than any global, plus I can't find any other test using globals - so I'm going with this as the easier option, but I'm not against revisiting properly supporting `env` later or in a follow-up PR if you'd like that (assuming it doesn't turn out to be needed anyway for this PR)

Reply:
> Next issue is that ESLint doesn't want to treat the imported files by this test as a `module`:

`named-exports.js` would be considered a module.  In the rule-tester compat, you could transform the `env` syntax to `ecmaVersion`  on parser options and map es6 to 2015?

Reply:
Looks like this is another bug similar to #3052 - if I apply that manually to v2.30.0 and then lint using a plain project I get the same error. 26eba2eb88847d1b374f8f0d603ddebeafdc9e04 seems to fix it, though I'm not 100% sure if there's a gotcha somewhere but we'll see what CI has to say 😄

Reply:
> the latest version of the globals package requires Node v18+.

Can't you ignore that and use the latest anyway? The `globals` JS code is just a tiny CommonJS wrapper than loads a JSON file. At the moment, you can still use it, as what it does is compatible with Node.js 4 (according to the last commit to that file): https://github.com/sindresorhus/globals/blob/main/index.js

Reply:
@targos unfortunately the `engines.node` declaration alone makes it a breaking change to include at runtime.

however, we could certainly *copy* the JSON file from that package in `prepack`, and then at least we have somewhat up-to-date info?

Reply:
And I think more importantly, as @G-Rath pointed out, there's not (yet) a need for it, since the only `env` usage today is the `es6` one, which is covered by `ecmaVersion`.  If more `env` props need to be supported by the rule tester compat in the future, that can be addressed then.